app/assets/javascripts/channels/synchronized_editor_channel.js

```diff
@@ -23,13 +24,18 @@ $(document).on('turbolinks:load', function () {
 
         received(data) {
           // Called when there's incoming data on the websocket for this channel
-          if (current_user_id !== data['current_user_id']) {
-            CodeOceanEditor.applyChanges(data['delta']['data']);
+          if (data.command === 'editor_change' && current_user_id !== data['current_user_id']) {
+              CodeOceanEditor.applyChanges(data['delta']['data'], data['active_file']);
+          } else if (data.command === 'connection_change' && current_user_id !== data.current_user_id) {
+              CodeOceanEditor.showUserIsOnline(data);
```

Comment:
Rather than passing the data, what if we would only pass the name and online status? Then, the method would be more independent of the actual data passed here.